Modules/_winapi.c

```diff
@@ -774,12 +774,172 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject*
+sortenvironmentkey(PyObject *module, PyObject *item)
+{
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT,
+                                      LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    "",
+};
+
+bool
+sort_environment_keys(PyObject *keys)
+{
+    PyObject *keyfunc = NULL, *kwnames = NULL;
+    bool result = false;
+
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    if (keyfunc == NULL) {
+        return false;
+    }
+    kwnames = Py_BuildValue("(s)", "key");
+    if (kwnames == NULL) {
+        goto cleanup;
+    }
+    PyObject *args[] = { keys, keyfunc };
+    PyObject *ret = PyObject_VectorcallMethod(&_Py_ID(sort), args, 1, kwnames);
+    if (ret == NULL) {
+        goto cleanup;
+    }
+    Py_DECREF(ret);
+
+    result = true;
+
+cleanup:
+    Py_XDECREF(keyfunc);
+    Py_XDECREF(kwnames);
```

Comment:
Post refactoring, this function is now pretty easy to handle, ref. count wise. IOW, I would use explicit decref's instead of `goto`s and a `cleanup` label.

Reply:
Done.